table.go

```diff
@@ -365,3 +365,17 @@ func (rt *RoutingTable) bucketIdForPeer(p peer.ID) int {
 	}
 	return bucketID
 }
+
+// maxCommonPrefix returns the maximum common prefix length between any peer in
+// the table and the current peer.
+func (rt *RoutingTable) maxCommonPrefix() uint {
+	rt.tabLock.RLock()
+	defer rt.tabLock.RUnlock()
+
+	for i := len(rt.buckets) - 1; i >= 0; i-- {
+		if rt.buckets[i].len() > 0 {
```

Comment:
https://github.com/libp2p/go-libp2p-kbucket/issues/72

Reply:
The number of buckets shouldn't matter in practice, right?

Reply:
The number of buckets we have shouldn't matter. We might care about the number of refreshes as long, but as long as it's ~15-20 and not ~100) it's probably fine.

I'm not sure if there's any tweaking to be done over how we do initial bootstrapping so that we fill our buckets fairly quickly. I suspect in the real network this will be much less problematic then the test network though.

Reply:
My point is that we:

1. Start with one big bucket.
2. Split the big bucket as we add peers.
3. Never shrink back down to 1 bucket.

In terms of _refreshing_ the routing table, the logic in this PR will refresh every CPL, not regardless of how many buckets we actually have. In practice, the number of buckets we have _shouldn't_ affect anything but memory usage, unless we're doing something wrong.